src/urllib3/util/retry.py

```diff
@@ -19,16 +18,21 @@
 from .util import reraise
 
 if TYPE_CHECKING:
+    from typing_extensions import Literal
+
     from urllib3.connectionpool import ConnectionPool
     from urllib3.response import HTTPResponse
 
 log = logging.getLogger(__name__)
 
 
 # Data structure for representing the metadata of requests that result in a retry.
-RequestHistory = namedtuple(
-    "RequestHistory", ["method", "url", "error", "status", "redirect_location"]
-)
+class RequestHistory(NamedTuple):
+    method: Optional[str]
+    url: Optional[str]
+    error: Optional[Exception]
+    status: Optional[int]
+    redirect_location: Optional[Union[str, "Literal[False]"]]
```

Comment:
When is this `False`? I think we don't add a `redirect_location` if `False` is given

Reply:
Yes, You are right. I changed it to `Optional[str]` and changed the `increment` function to handle it.